guides/17_Migration_to_3_0.md

```diff
@@ -163,6 +163,117 @@ handle them please remove.
+    console.log(`Receiving bandwidth is ${availableRecvBandwidth}, and old bandwidth is ${this.oldRecvBandwidth}`);
+  },
+  videoNotReceivingEnoughData: (receivingDataMap) => {
+    console.log(
+      `One or more video streams are not receiving expected amounts of data ${JSON.stringify(
+        Array.from(receivingDataMap.values())
+      )}`
+    );
+  }
+}
+// After
+const observer = {
+  oldSendBandwidthKbs: 0,
+  oldRecvBandwidthKbs: 0,
+  metricsDidReceive: (clientMetricReport) => {
+    const metricReport = clientMetricReport.getObservableMetrics();
+
+    const {
+      videoPacketSentPerSecond,
+      videoUpstreamBitrate,
+      nackCountPerSecond,
+    } = metricReport;
+    const availableSendBandwidthKbs = metricReport.availableOutgoingBitrate / 1000;
+    const availableRecvBandwidthKbs = metricReport.availableIncomingBitrate / 1000;
+    // videoSendHealthDidChange
+    console.log(`Sending video bitrate in kilobits per second: ${videoUpstreamBitrate / 1000} and sending packets per second: ${videoPacketSentPerSecond}`);
+
+    // videoSendBandwidthDidChange
+    if(this.oldSendBandwidthKbs != availableSendBandwidthKbs)
+    {
+      console.log(`Sending bandwidth is ${availableSendBandwidthKbs}, nack count per second is ${nackCountPerSecond}, and old bandwidth is ${this.oldSendBandwidthKbs}`);
+      this.oldSendBandwidthKbs = availableSendBandwidthKbs;
+    }
+
+    // videoReceiveBandwidthDidChange
+    if(this.oldRecvBandwidthKbs != availableRecvBandwidthKbs)
+    {
+      console.log(`Receiving bandwidth is ${availableRecvBandwidthKbs}, and old bandwidth is ${this.oldRecvBandwidthKbs}`);
+      this.oldRecvBandwidthKbs = availableRecvBandwidthKbs;
+    }
+
+    // videoNotReceivingEnoughData
+    const videoReceivingBitrateMap = new Map<string, ClientVideoStreamReceivingReport>();
+    let fireCallback = false;
+    for (const bitrate of this.currentAvailableStreamAvgBitrates.bitrates) {
+      if (downlinkVideoStream.has(bitrate.sourceStreamId)) {
+        const report = downlinkVideoStream.get(bitrate.sourceStreamId);
+        const attendeeId = this.context.videoStreamIndex.attendeeIdForStreamId(
+          bitrate.sourceStreamId
+        );
+        if (!attendeeId) {
+          continue;
+        }
+        const newReport = new ClientVideoStreamReceivingReport();
+        const prevBytesReceived = report.previousMetrics['bytesReceived'];
+        const currBytesReceived = report.currentMetrics['bytesReceived'];
+        if (!prevBytesReceived || !currBytesReceived) {
+          continue;
+        }
+
+        const receivedBitrate = ((currBytesReceived - prevBytesReceived) * 8) / 1000;
+
+        newReport.expectedAverageBitrateKbps = bitrate.avgBitrateBps / 1000;
+        newReport.receivedAverageBitrateKbps = receivedBitrate;
+        newReport.attendeeId = attendeeId;
+        if (
+          receivedBitrate <
+          (bitrate.avgBitrateBps / 1000) * MonitorTask.DEFAULT_DOWNLINK_CALLRATE_UNDERSHOOT_FACTOR
```

Comment:
Maybe we should not link to this `MonitorTask.DEFAULT_DOWNLINK_CALLRATE_UNDERSHOOT_FACTOR` as it is kinda internal variable inside sdk.

Reply:
This example will be removed